node/sc/sub_bridge_handler.go

```diff
@@ -389,28 +388,28 @@ func (sbh *SubBridgeHandler) updateTxCounts(block *types.Block) {
 	}
 
 	var startBlkNum uint64
-	if sbh.latestAnchoredBlockNumber == 0 {
+	if sbh.latestTxCountsAddedBlockNumber == 0 {
 		startBlkNum = block.NumberU64()
 	} else {
-		startBlkNum = sbh.latestAnchoredBlockNumber + 1
+		startBlkNum = sbh.latestTxCountsAddedBlockNumber + 1
 	}
 
 	for i := startBlkNum; i <= block.NumberU64(); i++ {
-		b := sbh.subbridge.blockchain.GetBlockByNumber(i)
-		if b == nil {
-			logger.Warn("blockAnchoringManager: break to generateAndAddAnchoringTxIntoTxPool by the missed block", "missedBlockNumber", i)
-			break
-		}
-		if b.NumberU64() >= sbh.txCountsEnabledBlockNumber {
+		if i >= sbh.txCountsEnabledBlockNumber {
+			b := sbh.subbridge.blockchain.GetBlockByNumber(i)
+			if b == nil {
+				logger.Warn("blockAnchoringManager: break to generateAndAddAnchoringTxIntoTxPool by the missed block", "missedBlockNumber", i)
+				break
+			}
 			sbh.txCounts += uint64(b.Transactions().Len())
 		}
 	}
+	sbh.UpdateLatestTxCountsAddedBlockNumber(block.NumberU64())
```

Comment:
How about moving this routine after 404:line to retry adding tx counts since the block.

Reply:
Good.